receiver/vcenterreceiver/metrics.go

```diff
@@ -278,6 +284,13 @@ func (v *vcenterMetricScraper) processHostPerformance(metrics []performance.Enti
 				case "disk.write.average":
 					v.mb.RecordVcenterHostDiskThroughputDataPoint(pcommon.NewTimestampFromTime(si.Timestamp), nestedValue, metadata.AttributeDiskDirectionWrite)
 				}
+				// Increment values of timestamps when finished with all nested values
+				if j == len(val.Value)-1 {
+					for ind, metricMapVal := range metricMap[val.Name] {
+						incrementedTime := time.Duration(metricMapVal.Interval) * time.Duration(j+1) * time.Nanosecond
```

Comment:
Can you help me understand what this is doing?

Is vcenter returning multiple data points with the same timestamp and no other differentiating information?

Reply:
Yes. processHostPerformance() goes through a map with a Value array (image 1)
<img width="314" alt="image" src="https://github.com/open-telemetry/opentelemetry-collector-contrib/assets/38274348/640e5448-39b5-43dc-85b4-9b124ce623c8">
The issue is that several Values of the same name
<img width="1017" alt="image" src="https://github.com/open-telemetry/opentelemetry-collector-contrib/assets/38274348/99ed9e97-8cc5-446b-8162-fc063c49b5b1">
will go through and report their datapoint values. There are 5 example timestamps that are supposed to go 1-1 with the length of the inner value it seems, but since they get reported multiple times, the timestamps get recycled. the above code you mentioned is just me keeping track and incrementing those timestamps so when they get encountered, its not being recycled. The differentiating factor between these values seems to be the Instance
<img width="818" alt="image" src="https://github.com/open-telemetry/opentelemetry-collector-contrib/assets/38274348/7d782ece-d5c7-4488-88c6-12ce2ef2a449">
but the recording of the data doesn't take the difference of the Instance into account.

Reply:
> but the recording of the data doesn't take the difference of the Instance into account.

That seems to be the problem. If I'm understanding correctly, we need to differentiate these data points with a resource attribute. In other words, we need to emit multiple resources, each with one of these data points.

@schmikei, WDYT?

Reply:
Sorry for the delay, was on vacation!

It feels like this is best solved by adding a `vcenter.host.disk.instance` attribute or something. What's being reported from the PerformanceManager API for this particular host seems to have 2 drives and that is the crux of the case you're showing right now. Seems like that is the approach I think makes the most sense @BominRahmani messing with the timestamps seems like the incorrect approach.